File: test/fixtures/ruby/options.rb
require 'net/http'

uri = URI('http://localhost:28139/api/tunein/queue-and-play')
params = {
  :deviceSerialNumber => 'xxx',
  :deviceType => 'xxx',
  :guideId => 's56876',
  :contentType => 'station',
  :callSign => '',
  :mediaOwnerCustomerId => 'xxx',
}
uri.query = URI.encode_www_form(params)

req = Net::HTTP::Options.new(uri)
req['Pragma'] = 'no-cache'
req['Access-Control-Request-Method'] = 'POST'
req['Origin'] = 'https://alexa.amazon.de'
# req['Accept-Encoding'] = 'gzip, deflate, br'
req['Accept-Language'] = 'de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4'
req['User-Agent'] = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36'
req['Accept'] = '*/*'
req['Cache-Control'] = 'no-cache'
req['Referer'] = 'https://alexa.amazon.de/spa/index.html'
req['Connection'] = 'keep-alive'
req['DNT'] = '1'
req['Access-Control-Request-Headers'] = 'content-type,csrf'

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/java/options.java
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

HttpClient client = HttpClient.newHttpClient();

HttpRequest request = HttpRequest.newBuilder()
    .uri(URI.create("http://localhost:28139/api/tunein/queue-and-play?deviceSerialNumber=xxx&deviceType=xxx&guideId=s56876&contentType=station&callSign=&mediaOwnerCustomerId=xxx"))
    .method("OPTIONS", HttpRequest.BodyPublishers.noBody())
    .setHeader("Pragma", "no-cache")
    .setHeader("Access-Control-Request-Method", "POST")
    .setHeader("Origin", "https://alexa.amazon.de")
    .setHeader("Accept-Encoding", "gzip, deflate, br")
    .setHeader("Accept-Language", "de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4")
    .setHeader("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36")
    .setHeader("Accept", "*/*")
    .setHeader("Cache-Control", "no-cache")
    .setHeader("Referer", "https://alexa.amazon.de/spa/index.html")
    .setHeader("Connection", "keep-alive")
    .setHeader("DNT", "1")
    .setHeader("Access-Control-Request-Headers", "content-type,csrf")
    .build();

HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
